presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlock.java

```diff
@@ -158,6 +158,11 @@ public Block copyRegion(int positionOffset, int length)
     {
         checkValidRegion(getPositionCount(), positionOffset, length);
 
+        if (positionOffset == 0 && length == positionOffset &&
+                arrayOffset == 0 && positionCount == values.length) {
+            return this;
```

Comment:
A `Block` is a mutable data structure. Is it obvious that it's safe to return non-copy from `copyRegion` method? _If it is_, consider renaming the method to avoid confusion in the future.